src/game.cpp

```diff
@@ -9948,7 +9948,10 @@ void game::wield( item_location &loc )
         add_msg( m_info, "%s", ret.c_str() );
     }
 
-    u.wield( u.i_at( loc.obtain( u ) ) );
+    // Can't use loc.obtain() here because that would cause things to spill
+    u.moves -= loc.obtain_cost( u );
```

Comment:
`u.mod_moves( -loc.obtain_cost( u ) );` would be slightly better (`moves` should eventually be encapsulated at some point).